Python/pystate.c

```diff
@@ -1746,6 +1747,47 @@ PyGILState_Release(PyGILState_STATE oldstate)
         PyEval_SaveThread();
 }
 
+int
+PyThread_TryAcquireFinalizeBlock()
+{
+    _PyRuntimeState *runtime = &_PyRuntime;
+    MUTEX_LOCK(runtime->ceval.gil.mutex)
+    if (runtime->finalize_blocks & 1) return 0;
```

Comment:
you don't need to unlock the mutex in the return case here ?

Reply:
Thanks --- fixed.